x-pack/plugin/esql-core/src/main/java/org/elasticsearch/xpack/esql/core/expression/TypeResolutions.java

```diff
@@ -85,10 +86,10 @@ public static TypeResolution isRepresentableExceptCountersDenseVectorAndAggregat
     ) {
         return isType(
             e,
-            dt -> isRepresentable(dt) && dt != DENSE_VECTOR && dt != AGGREGATE_METRIC_DOUBLE,
+            dt -> isRepresentable(dt) && dt != DENSE_VECTOR && dt != AGGREGATE_METRIC_DOUBLE && dt != DATE_RANGE,
```

Comment:
This seems to exclude this type from many functions. Is that because we plan to add that support in followup PRs?

Reply:
Probably worth a comment that this is temporary while under construction.

I'm not 100% sure why we need this to be honest though - I had thought we didn't try sending these "under construction" types in tests already.